src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -180,10 +184,52 @@ MulticopterAttitudeControl::generate_attitude_setpoint(const Quatf &q, float dt,
 	// Align the desired tilt with the yaw setpoint
 	Quatf q_sp = q_sp_yaw * q_sp_rp;
 
-	q_sp.copyTo(attitude_setpoint.q_d);
-
 	// Transform to euler angles for logging only
+	Eulerf euler_des(q_sp);
+
+	/*** CUSTOM ***/
+
+	tilting_servo_sp_s servo_sp;
+
+	if(_param_airframe.get() == 12 ){ //If tilting_multirotors
```

Comment:
Probably we should have a completely distinct attitude generation in a separate function for this use case.

Reply:
I can write a function with the same instructions and then call it to make it more easy to understand. Is that what you mean?